### PR TITLE
Introduce command changes in the AoC Cog

### LINKED ISSUE #325: Personal stats with something like !aoc mystats
Instead of needing to go on the website to check your stats, you can use the bot!

## Implementation details
- Show your placement in the private leaderboard
- Show the stars you attained
- Show points that you have
- Suggestions?

- [ ] Write the bot command
- [ ] Add the command to our README.md

## Additional information
You can find your own stats here: https://adventofcode.com/2019/leaderboard/private/view/631135
by doing `ctrl-f`. But you can also look at your personal global stats here: https://adventofcode.com/2019/leaderboard/self

There are the best places to look when finding things to put on the embed.

### PR DESCRIPTION
**Pull request by D0rs4n**
## Relevant Issues

Closes #325 

## Description
- The AoC day and star browser has been separated from the leaderboard command, from now on it's a separate command
- The leaderboard command has a new `self_placement_name` option, that shows the personal stats for the specified profile name.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/events/advent_of_code/_cog.py [hunk 1]
```diff
@@ -235,6 +216,43 @@ async def aoc_leaderboard(
...
+    @whitelist_override(channels=AOC_WHITELIST_RESTRICTED)
+    async def aoc_leaderboard(
+            self,
+            ctx: commands.Context,
+            self_placement_name: Optional[str] = None,
+    ) -> None:
+        """
+        Get the current top scorers of the Python Discord Leaderboard.
+
+        Additionaly you can specify a `self_placement_name`
```
Comment (jchristgit) [inline bot/exts/events/advent_of_code/_cog.py:234]: Small typo:

```suggestion
        Additionally you can specify a `self_placement_name`
```

**Review by jchristgit — APPROVED** [review]
Small typo. Apart from that, very clean code, as always from you!

Maybe a minor suggestion: possibly we could make the code in `_format_leaderboard` a _tiny_ bit cleaner by moving out the line formatting into a separate function. Specifically this bit:

```py
                AOC_TABLE_TEMPLATE.format(
                     rank=rank,
                     name=f"(You) {data['name']}",
                     score=str(data["score"]),
                     stars=f"({data['star_1']}, {data['star_2']})"
                 )
```
It's only repeated twice, so I don't think it's a dealbreaker. I do think it would simplify things though. Maybe something like:

```py
def _format_leaderboard_line(rank: int, data: dict[str, Any], *, is_author: bool) -> str:
    """Build a string representing a line of the leaderboard.

    Parameters:
        rank:
            Rank in the leaderboard of this entry.

        data:
            Mapping with entry information.

    Keyword arguments:
        is_author:
            Whether to address the name displayed in the returned line
            personally.

    Returns:
        A formatted line for the leaderboard.

    Notes:
        I copy and pasted this code.
    """

    return AOC_TABLE_TEMPLATE.format(
        rank=rank,
        name=data['name'] if not is_author else f"(You) {data['name']}",
        score=str(data['score']),
        stars=f"({data['star_1']}, {data['star_2']})"
    )
```

File: bot/exts/events/advent_of_code/_cog.py (hunk 1 above)
Comment (D0rs4n) [reply bot/exts/events/advent_of_code/_cog.py:234]: Thank you for the quick review! 
Sorry, I should've made the `_format_leaderboard`  more consistent.
Will push a patch ASAP.

Comment (D0rs4n) [reply bot/exts/events/advent_of_code/_cog.py:234]: Also, I noticed a minor problem, in the code.
I have to modify the cache a little bit. 
Should be fine!

**Comment by D0rs4n** [comment]
Update: Yes, there is a problem with caching, I've found a solution and will push a change along with the others.

**Comment by D0rs4n** [comment]
The issue has been patched, everything should work as expected! 👍

**Review by Akarys42 — APPROVED** [review]
Lgtm

**Pull request merged**